src/heaps/fibonacci_heap.jl

```diff
@@ -0,0 +1,367 @@
+end
+
+MutableFibonacciHeapNode(data::T) where T = MutableFibonacciHeapNode{T}(data, 0, nothing, nothing, nothing, nothing, false, false)
+#################################################
+#
+#   heap type and constructors
+#
+#################################################
+mutable struct MutableFibonacciHeap{T} <: AbstractMutableHeap{T}
+    rootList::Union{MutableFibonacciHeapNode{T}, Nothing}
+    minNode::Union{MutableFibonacciHeapNode{T}, Nothing}
+    nodeCount::Int
+
+    MutableFibonacciHeap{T}() where T = new{T}(nothing, nothing, 0)
+
+    function MutableFibonacciHeap{T}(xs::AbstractVector{T}) where T
+        fiboHeap = _make_fibonacci_heap(xs,T)
+        new{T}(fiboHeap.rootList,fiboHeap.minNode,fiboHeap.nodeCount)
+    end
+    
+end
+const FibonacciHeap{T} = MutableFibonacciHeap{T}
+FibonacciHeap(xs::AbstractVector{T}) where T = FibonacciHeap{T}(xs)
+#################################################
+#
+#   core implementation
+#
+#################################################
+# make fibonacci_heap from Vector
+function _make_fibonacci_heap(xs::AbstractVector{T},Typ::Type{T}) where T
+    len = length(xs)
+    fiboHeap = MutableFibonacciHeap{Typ}()
+    for i = 1 : len
+        push!(fiboHeap,xs[i])
+    end
+    fiboHeap
+end
+
+# merge a node with the doubly linked root list
+function _merge_with_root_list(root::MutableFibonacciHeap{T}, node::MutableFibonacciHeapNode{T}) where T
+    if root.rootList == nothing
+        root.rootList = node
+    else
+        node.right = root.rootList.right
+        node.left = root.rootList
+        root.rootList.right.left = node
+        root.rootList.right = node
+    end
+end
+# remove a node from the doubly linked root list
+function _remove_from_root_list(root::MutableFibonacciHeap{T}, node::MutableFibonacciHeapNode{T}) where T
+    if node == root.rootList
+        root.rootList = node.right
+    end
+    node.left.right = node.right
+    node.right.left = node.left
+end
+
+# remove a node from the doubly linked child list
+function _remove_from_child_list(root::MutableFibonacciHeap{T}, parent::MutableFibonacciHeapNode{T}, node::MutableFibonacciHeapNode{T}) where T
+    if parent.child == parent.child.right
+        parent.child = nothing
+    elseif parent.child == node
+        parent.child = node.right
+        node.right.parent = parent
+    end
+    node.left.right = node.right
+    node.right.left = node.left
+end
+
+# merge a node with the doubly linked child list of a root node
+function _merge_with_child_list(root::MutableFibonacciHeap{T}, parent::MutableFibonacciHeapNode{T}, node::MutableFibonacciHeapNode{T}) where T
+    if parent.child == nothing
+        parent.child = node
+    else
+        node.right = parent.child.right
+        node.left = parent.child
+        parent.child.right.left = node
+        parent.child.right = node
+    end
+end
+
+# extract (delete) the min node from the heap in O(log n) time
+function extractMin(root::MutableFibonacciHeap)
```

Comment:
This PR has a mix of camelCase and snake_case. Julia's style guide (and the rest of DataStructures, as far as I know) uses snake_case, so that is preferable.